packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4186,18 +4202,28 @@ class InheritedElement extends ProxyElement {
   /// Calls [Element.didChangeDependencies] of all dependent elements, if
   /// [InheritedWidget.updateShouldNotify] returns true.
   ///
-  /// Notifies all dependent elements that this inherited widget has changed.
+  /// Called by [update], immediately prior to [build].
   ///
-  /// [InheritedElement] calls this function if the [widget]'s
-  /// [InheritedWidget.updateShouldNotify] returns true.
+  /// Calls [notifyClients] to actually trigger the notifications.
+  @override
+  void updated(InheritedWidget oldWidget) {
+    if (widget.updateShouldNotify(oldWidget))
+      super.updated(oldWidget);
+  }
+
+  /// Notifies all dependent elements that this inherited widget has changed, by
+  /// calling [Element.didChangeDependencies].
+  ///
+  /// This method must only be called during the build phase. Usually this
+  /// method is called automatically when an inherited widget is rebuilt, e.g.
+  /// as a result of calling [State.setState] above the inherited widget.
   ///
-  /// This method must be called during the build phase. Usually this method is
-  /// called automatically when an inherited widget is rebuilt, e.g. as a
-  /// result of calling [State.setState] above the inherited widget.
+  /// See also:
+  ///
+  ///  * [InheritedNotifier], a subclass of [InheritedWidget] that also calls
+  ///    this method when its [Listenable] sends a notification.
   @override
   void notifyClients(InheritedWidget oldWidget) {
-    if (!widget.updateShouldNotify(oldWidget))
```

Comment:
Doing this test in `updated` is subtly different (since calls to notifyClients used to be gated by updateShouldNotify) Hopefully no one will notice.

Reply:
Yeah, I wrestled with this quite a bit. I hope this is a subtle enough change to be fine.